boards/arm/sam4s_xplained/ext_ram.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Justin Watson 2017
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <init.h>
+#include <nanokernel.h>
+#include <pinmux.h>
+#include <soc.h>
+#include <sys_io.h>
+
+/* Static Memory Controller (SMC) */
```

Comment:
The way the code is written is indeed pretty board / memory specific. No offense to anyone but it feels like a quick hack. I believe we should have a proper implementation with a standalone SMC driver. The SMC module is pretty complex, apart from SRAM it can support PSRAM, PROM, EPROM, EEPROM, LCD Module, NOR Flash and NAND Flash. For now the driver code can be very simple, but if it's a standalone code we will have a place where it can grow when the need arises.

I still think it would be better to add external RAM support in a separate patch. It seems like the topic needs some more thought.